doc/api/assert.md

```diff
@@ -147,6 +147,142 @@ try {
 }
 ```
 
+## Class: assert.CallTracker
+
+### `new assert.CallTracker()`
+<!-- YAML
+added: REPLACEME
+-->
+
+Creates a new [`CallTracker`][] object which can be used to track if functions
+were called a specific number of times. The `tracker.verify()` must be called
+for the verification to take place. The usual pattern would be to call it in the
+`process` `exit` handler.
+
+```js
+const assert = require('assert');
+
+const tracker = new assert.CallTracker();
+
+function func() {}
+
+// callsfunc() must be called exactly 1 time before tracker.verify()
+const callsfunc = tracker.calls(func, 1);
+
+callsfunc();
+
+process.on('exit', () => {
+  // Verifies if all tracker.calls() functions have been called exact times
+  tracker.verify();
+});
+```
+
+### `assert.CallTracker.calls([fn][, exact])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `fn` {Function} **Default** A no-op function.
+* `exact` {number} **Default** `1`.
+* Returns: {Function} that wraps `fn`.
+
+The wrapper function is expected to be called exactly `exact` times. If the
+function is not called exactly `exact` times before
+[`assert.CallTracker.verify()`][] is called, an exception is thrown.
+
+```js
+const assert = require('assert');
+
+// Creates call tracker.
+const tracker = assert.createCallTracker();
+
+function func() {}
+
+// Returns a function that wraps func() that must be called exact times
+// before tracker.verify().
+const callsfunc = tracker.calls(func);
+```
+
+### `assert.CallTracker.report()`
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Array} of objects containing information about the wrapper functions
+returned by [`assert.CallTracker.calls()`][].
+
+The arrays contains information about the expected and actual number of calls of
+the functions that have not been called the expected number of times.
+
+```js
+const assert = require('assert');
+
+// Creates call tracker.
+const tracker = assert.createCallTracker();
+
+function func() {}
+
+function foo() {}
+
+// Returns a function that wraps func() that must be called exact times
+// before tracker.verify().
+const callsfunc = tracker.calls(func);
+
+const callsfoo = tracker.calls(foo);
+
+// Returns an array containing error objects for func() and foo()
+tracker.report();
+```
+
+### `assert.CallTracker.verify()`
+<!-- YAML
+added: REPLACEME
+-->
+
+Iterates through the list of functions passed to
+[`assert.CallTracker.calls()`][] and will throw an error for functions that
+have not been called the expected number of times.
+
+```js
+const assert = require('assert');
+
+// Creates call tracker.
+const tracker = assert.createCallTracker();
+
+function func() {}
+
+// Returns a function that wraps func() that must be called exact times
+// before tracker.verify().
+const callsfunc = tracker.calls(func, 2);
+
+callsfunc();
+
+// Will throw an error since callsfunc() was only called once.
+tracker.verify();
+```
+
+### `assert.CallTracker.verifyOnExit()`
```

Comment:
Is this truly need? The implementation is a oneliner so it's not really simplifying anything.
I would prefer to have it as an option in `createCallTracker` instead, e.g. `createCallTracker({ onExit: true })`.

Reply:
Hmm yes it doesn't really simplify it that much. I like the idea of making it a constructor parameter